Sources/Sharing/Internal/Reference.swift

```diff
@@ -529,15 +529,15 @@ extension _ManagedReference: MutableReference, Equatable where Key: SharedKey {
   }
 }
 
-final class _AppendKeyPathReference<
```

Comment:
Let's leave the current hierarchy as is. While tempting to abstract here, for now I think it's a little easier to maintain as a brand new `_ReadClosureReference` conformance.